packages/web-components/fast-components/src/accordion/accordion.ts

```diff
@@ -0,0 +1,132 @@
+import { attr, FASTElement, observable } from "@microsoft/fast-element";
+import {
+    keyCodeArrowDown,
+    keyCodeArrowUp,
+    keyCodeEnd,
+    keyCodeHome,
+    wrapInBounds,
+} from "@microsoft/fast-web-utilities";
+import { StartEnd } from "../patterns/start-end";
+import { applyMixins } from "../utilities";
+import { FASTAccordionItem } from "./accordion-item";
+
+export enum AccordionExpandMode {
+    single = "single",
+    multi = "multi",
+}
+
+export class Accordion extends FASTElement {
+    @attr
+    public expandmode: AccordionExpandMode = AccordionExpandMode.multi;
+
+    @observable
+    public accordionItems: HTMLElement[];
+    public accordionItemsChanged(): void {
+        if (this.$fastController.isConnected) {
+            this.accordionIds = this.getHeaderIds();
+            this.setItems();
+        }
+    }
+
+    @observable
+    public activeIndicatorRef: HTMLElement;
+
+    private activeid: string;
+    private activeHeaderIndex: number = 0;
+    private accordionIds: Array<string | null>;
+
+    private change = (): void => {
+        this.$emit("change", this.activeid);
+    };
+
+    private setItems = (): void => {
+        this.accordionIds = this.getHeaderIds();
+        this.accordionItems.forEach((header: HTMLElement, index: number) => {
+            if (header instanceof FASTAccordionItem) {
```

Comment:
@nicholasrice Did we determine that this `instanceof` check can fail, even if the type is imported (and registered) first?

Reply:
@nicholasrice and I chatted about this, I'll let him chime in. But think we should be good here.

Reply:
We determined that import-order matters in cases where DOM exists prior to script parsing / execution. In those cases, the elements are upgraded immediately upon definition. That's why we were seeing child components of the DS provider upgrading before the DS provider when the DS provider was imported and referenced *after* the child components. I *think* that case is covered here because the item is an explicit dependency of the accordion.